### PR TITLE
Navbar and Footer

### PR DESCRIPTION
**Pull request by ShawnAviles**
# Description

Completed implementation for basic Navbar and Footer:
- Navbar includes what was listed in the ticket (Sign in, Sign out, Directories, Profile, Settings, Home) and whatever currently exists for Admin
  - when not sign in, there is no Avatar just a "Sign in" button 
- Footer mimics current footer on AAD live website

Additional features:
- ensured that `AdminLayout` is only viewable by admin (other roles cannot see it anymore)
- removed additional buttons from home and profile page as those buttons live in the Navbar now 

Admin:
Screenshot 2024-03-27 at 3 46 32 PM

Mentee / Mentor:
Screenshot 2024-03-27 at 3 47 01 PM

Footer: 
Screenshot 2024-03-27 at 3 47 17 PM

Related #147

### CODE REVIEW

File: @types/nextjs-routes.d.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ declare module "next/link" {
   export interface LinkProps
     extends Omit<NextLinkProps, "href" | "locale">,
       AnchorHTMLAttributes<HTMLAnchorElement> {
-    href: Route | StaticRoute | Omit<Route, "pathname">
+    href: Route | StaticRoute | Omit<Route, "pathname"> | DynamicRoute;
```
Comment (ShawnAviles) [inline @types/nextjs-routes.d.ts:103]: Note: This was needed for using the Next Link Component in the Navbar. That implementation was decided upon via [this post](https://stackoverflow.com/questions/32106513/material-ui-menu-using-routes). Bc we could use the MUI Button Component but limits our custom styling options.

Comment (ulrokx) [reply @types/nextjs-routes.d.ts:103]: Spoke offline, this file should not be modified and the changes can be made in Navbar component

File: src/components/layouts/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+import MenuIcon from "@mui/icons-material/Menu";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import MenuItem from "@mui/material/MenuItem";
+import AvatarWrapper from "../settings/AvatarWrapper";
+import { signIn, signOut, useSession } from "next-auth/react";
+import Link from "next/link";
+
+const pagesMenteeAndMentor = [
```
Comment (ulrokx) [inline src/components/layouts/Navbar.tsx:17]:
```tsx
interface NavbarLink {
  name: string;
  path: Route["pathname"];
}
```

Comment (ulrokx) [reply src/components/layouts/Navbar.tsx:17]: Give both of these arrays type `NavbarLink[]`

File: src/components/layouts/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+                    transformOrigin={{
+                      vertical: "top",
+                      horizontal: "right",
+                    }}
+                    open={Boolean(anchorElUser)}
+                    onClose={handleCloseUserMenu}
+                  >
+                    <MenuItem
+                      component={Link}
+                      href={`/profiles/${sessionData?.user?.id}`}
```
Comment (ulrokx) [inline src/components/layouts/Navbar.tsx:215]:
```tsx
href={{
    pathname: "/profiles/[id]",
    query: { id: sessionData?.user?.id },
}}
```

Comment (ulrokx) [reply src/components/layouts/Navbar.tsx:215]: Taken from https://www.npmjs.com/package/nextjs-routes#examples-

File: src/pages/_app.tsx [hunk 4]
```diff
@@ -29,7 +31,11 @@ const MyApp = ({
   return (
     <SessionProvider session={session}>
       <StyledEngineProvider injectFirst>
-        {getLayout(<Component {...pageProps} />)}
+        {getLayout(
+          <RootLayout>
```
Comment (ulrokx) [inline src/pages/_app.tsx:35]: I think RootLayout should wrap the getLayout function, not the other way around

**Review by ulrokx — CHANGES_REQUESTED** [review]
Also discussed offline, admin sidebar is only on admin page, not home page for admins

File: src/pages/_app.tsx (hunk 4 above)
Comment (ShawnAviles) [reply src/pages/_app.tsx:35]: Got it, was confused on the intention before. Will update this as well as only have the admin layout show the admin page.

**Review by ulrokx — APPROVED** [review]

**Pull request merged**